src/installer/scripts.py

```diff
@@ -43,6 +43,41 @@
 """
 
 
+def _is_executable_simple(executable: bytes) -> bool:
+    if b" " in executable:
+        return False
+    shebang_length = len(executable) + 3  # Prefix #! and newline after.
+    # According to distlib, Darwin can handle up to 512 characters. But I want
+    # to avoid platform sniffing to make this as platform-agnostic as possible.
+    # The "complex" script isn't that bad anyway.
+    return shebang_length <= 127
+
+
+def _build_shebang(executable: str, forlauncher: bool) -> bytes:
+    """Build a shebang line.
+
+    The non-launcher cases are taken directly from distlib's implementation,
+    which tries its best to account for command length, spaces in path, etc.
+
+    https://bitbucket.org/pypa/distlib/src/58cd5c6/distlib/scripts.py#lines-124
+    """
+    executable_bytes = executable.encode("utf-8")
+    if forlauncher:  # The launcher can just use the command as-is.
+        return b"#!" + executable_bytes
+    if _is_executable_simple(executable_bytes):
+        return b"#!" + executable_bytes
```

Comment:
I have just noticed, that for both those cases we return exactly the same value. Can you make it into a single `if` statement with an `or`?

Reply:
I have a followup commit that does this. It's not really appropriate when moving a block of code from one location to another, to also start changing the interior design of that block.